### PR TITLE
bump NDK to latest LTS or 25

### PR DESCRIPTION
**Pull request by dulmandakh**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -26,7 +26,7 @@ ENV DEBIAN_FRONTEND=noninteractive
 ARG SDK_VERSION=commandlinetools-linux-8512546_latest.zip
 ARG ANDROID_BUILD_VERSION=31
 ARG ANDROID_TOOLS_VERSION=31.0.0
-ARG NDK_VERSION=21.4.7075529
+ARG NDK_VERSION=25.0.8775105
```
Comment (cortinico) [inline Dockerfile:29]: This should be aligned with a change on React Native's end (especially the template)

Comment (gengjiawen) [reply Dockerfile:29]: CI also looks failed:

```
 /opt/android/ndk/25.0.8775105/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64 doesn't exist.
```

Comment (dulmandakh) [reply Dockerfile:29]: Buck lacks support for NDK 21+ 😢

Comment (leotm) [reply Dockerfile:29]: it'll just be updating [android/build.gradle](https://github.com/facebook/react-native/blob/54a4fcbfdcc8111b3010b2c31ed3c1d48632ce4c/template/android/build.gradle#L12) right

NDK 25 been looking pretty solid for me past few months

think only extra thing had to do was bump CMake from 3.22.1 -> [3.23.1](https://github.com/leotm/react-native-template-new-architecture/pull/585/files#diff-0d634959c8276ae976e39ba475e63d0b9ec27824470c79f1971f58fe37c46325) on [Ubuntu 20.04](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md)

Comment (cortinico) [reply Dockerfile:29]: Yup I think we should be fine by updating everything to NDK 25 for RN 0.71

Comment (dulmandakh) [reply Dockerfile:29]: Thanks for the comments. I'll try to finish the PR very soon.

**Pull request closed without merging**